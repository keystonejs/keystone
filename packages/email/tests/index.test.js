// NOTE: We don't test the `.send()` functionality as we assume keystone-email
// has tested it and it works.
describe('email senders', () => {
  test('jsx', async () => {
    const { emailSender } = require('../');

    const jsxEmailSender = emailSender.jsx({ root: `${__dirname}/jsx-views` });
    const { html, text } = await jsxEmailSender('view.jsx').render({ name: 'Foo' });

    expect(text).toEqual('Hello Foo');
    expect(html).toEqual('<!DOCTYPE html><html><body><div>Hello Foo</div></body></html>');
  });

  test('pug', async () => {
    const { emailSender } = require('../');

    const pugEmailSender = emailSender.pug({ root: `${__dirname}/pug-views` });
    const { html, text } = await pugEmailSender('view.pug').render({ name: 'Foo' });

    expect(text).toEqual('Foo is Awesome');
    expect(html).toEqual('<!DOCTYPE html><html><body><div>Foo is Awesome</div></body></html>');
  });

  test('mjml', async () => {
    const { emailSender } = require('../');

    const mjmlEmailSender = emailSender.mjml({ root: `${__dirname}/mjml-views` });
    const { html, text } = await mjmlEmailSender('view.jsx').render({ name: 'Foo' });

    expect(text).toMatchSnapshot();
    expect(html).toMatchSnapshot();
  });
});
