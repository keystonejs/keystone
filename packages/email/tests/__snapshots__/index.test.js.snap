// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email senders mjml 1`] = `
"Special Offer Inside![https://static.wixstatic.com/media/5cb24728abef45dabebe7edc1d97ddd2.jpg]
Get offer now! [https://www.example.com/]"
`;

exports[`email senders mjml 2`] = `
"<!doctype html><html xmlns=\\"http://www.w3.org/1999/xhtml\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\"><head><title>Foo Last Minute Offer For You!</title><!--[if !mso]><!-- --><meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\"><!--<![endif]--><meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1\\"><style type=\\"text/css\\">#outlook a { padding:0; }
          body { margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%; }
          table, td { border-collapse:collapse;mso-table-lspace:0pt;mso-table-rspace:0pt; }
          img { border:0;height:auto;line-height:100%; outline:none;text-decoration:none;-ms-interpolation-mode:bicubic; }
          p { display:block;margin:13px 0; }</style><!--[if mso]>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        <![endif]--><!--[if lte mso 11]>
        <style type=\\"text/css\\">
          .outlook-group-fix { width:100% !important; }
        </style>
        <![endif]--><!--[if !mso]><!--><link href=\\"https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700\\" rel=\\"stylesheet\\" type=\\"text/css\\"><style type=\\"text/css\\">@import url(https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700);</style><!--<![endif]--><style type=\\"text/css\\">@media only screen and (min-width:480px) {
        .mj-column-per-100 { width:100% !important; max-width: 100%; }
      }</style><style type=\\"text/css\\">@media only screen and (max-width:480px) {
      table.full-width-mobile { width: 100% !important; }
      td.full-width-mobile { width: auto !important; }
    }</style></head><body><div style=\\"display:none;font-size:1px;color:#ffffff;line-height:1px;max-height:0px;max-width:0px;opacity:0;overflow:hidden;\\">Special Offer Inside!</div><div><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:#efefef;background-color:#efefef;width:100%;\\"><tbody><tr><td><!--[if mso | IE]><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" style=\\"width:500px;\\" width=\\"500\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]--><div style=\\"margin:0px auto;max-width:500px;\\"><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\"><tbody><tr><td style=\\"direction:ltr;font-size:0px;padding:20px 0;text-align:center;\\"><!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:500px;\\" ><![endif]--><div class=\\"mj-column-per-100 outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"><table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"><tr><td align=\\"center\\" style=\\"font-size:0px;padding:10px 25px;word-break:break-word;\\"><table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:collapse;border-spacing:0px;\\"><tbody><tr><td style=\\"width:450px;\\"><img height=\\"auto\\" src=\\"https://static.wixstatic.com/media/5cb24728abef45dabebe7edc1d97ddd2.jpg\\" style=\\"border:0;display:block;outline:none;text-decoration:none;height:auto;width:100%;font-size:13px;\\" width=\\"450\\"></td></tr></tbody></table></td></tr></table></div><!--[if mso | IE]></td></tr></table><![endif]--></td></tr></tbody></table></div><!--[if mso | IE]></td></tr></table><![endif]--></td></tr></tbody></table><!--[if mso | IE]><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" style=\\"width:500px;\\" width=\\"500\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]--><div style=\\"margin:0px auto;max-width:500px;\\"><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\"><tbody><tr><td style=\\"direction:ltr;font-size:0px;padding:20px 0;text-align:center;\\"><!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:500px;\\" ><![endif]--><div class=\\"mj-column-per-100 outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"><table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"><tr><td align=\\"center\\" vertical-align=\\"middle\\" style=\\"font-size:0px;padding:20px;word-break:break-word;\\"><table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:separate;line-height:100%;\\"><tr><td align=\\"center\\" bgcolor=\\"#346DB7\\" role=\\"presentation\\" style=\\"border:none;border-radius:3px;cursor:auto;mso-padding-alt:10px 25px;background:#346DB7;\\" valign=\\"middle\\"><a href=\\"https://www.example.com/\\" style=\\"display:inline-block;background:#346DB7;color:#ffffff;font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;font-weight:normal;line-height:120%;margin:0;text-decoration:none;text-transform:none;padding:10px 25px;mso-padding-alt:0px;border-radius:3px;\\" target=\\"_blank\\">Get offer now!</a></td></tr></table></td></tr></table></div><!--[if mso | IE]></td></tr></table><![endif]--></td></tr></tbody></table></div><!--[if mso | IE]></td></tr></table><![endif]--></div></body></html>"
`;
